homeassistant/core.py

```diff
@@ -431,12 +431,13 @@ def __init__(self, config_dir: str) -> None:
         self.import_executor = InterruptibleThreadPoolExecutor(
             max_workers=1, thread_name_prefix="ImportExecutor"
         )
+        self._loop_thread_id = (
+            self.loop.__dict__.get("_thread_ident") or threading.get_ident()
```

Comment:
Is the optimization of calling `loop.__dict__` still necessary now that it happens in the constructor, which runs at the very beginning of startup

Reply:
I think we can drop that.

I need to investigate the failures. I think we might have some legitimate issues there.

Reply:
It was a problem in the test mocking.

I removed the `__dict__` call and make it fallback to the asyncio loop thread_id if ours is missing so its never not set.